src/widget/wbattery.cpp

```diff
@@ -86,7 +86,7 @@ QString formatMinutes(int minutes) {
 
 int pixmapIndexFromPercentage(double dPercentage, int numPixmaps) {
     // See WDisplay::getActivePixmapIndex for more info on this.
-    int result = static_cast<int>(dPercentage * numPixmaps - 0.00001);
+    int result = static_cast<int>(dPercentage * numPixmaps - 0.00001) / 100;
```

Comment:
Looks reasonable. We should then also remove ` - 0.00001`, because it is not necessary due to the following line.

I would prefer to apply the static cast to the whole expression instead of using an integer division:

```
int result = static_cast<int>(dPercentage / 100.0 * numPixmaps);
```

The literal `100.0` instead of `100` reassures that we are performing the calculation in floating-point. The result should be the same.

Reply:
agreed